packages/flame_svg/lib/svg.dart

```diff
@@ -37,6 +47,28 @@ class Svg {
   ) {
     canvas.renderAt(position, (c) => render(c, size));
   }
+
+  Image? _getImage(Vector2 size) {
+    final image = _imageCache.getValue(size);
```

Comment:
I am not sure why wouldn't it work well? I tested and it seems fine

Reply:
There could be two potential problems here (and perhaps we could add simple tests for them):
1. If you place an image into the cache using `Vector2.all(10)` as a size, then try to retrieve an image with size `Vector2.all(10)`, you won't find anything because Vector2 objects compare by identity and not by values.
2. If you have an image of `size=Vector2.all(10)` placed into the cache, and then later on you modify `size` in-place for example via `size.scale(10)`, then using `size` to retrieve the image from the cache will find the stored image even though that stored image was computed for size 10 not for size 100.

PS: `Size` may be more idiomatic than `Offset` here.

Reply:
I don't believe this is true, I went to check on the docs and it does define the == operator https://api.flutter.dev/flutter/vector_math_64/Vector2/operator_equals.html

Reply:
I stand corrected then, must have confused with another language.

Reply:
Still, if the purpose of the cache is to hold multiple versions of the image rendered at different resolutions, then it may not work properly.

For example, suppose you have an svg image of size `size0 = Vector2(10,10)`. You store `cache[size0] = image0` into the cache. Then you increase the size: `size0.scale(10)` and save the new image into the cache: `cache[size0] = image1`. At this point if you'd try to retrieve `cache[Vector2(10,10)]` the cache wouldn't be able to find it because the hash of the stored key has diverged from the key's equality identity.

Reply:
In the end, the fact that the vector can be internally mutated was indeed a problem, I changed to Size.